command/seal_migration_test.go

```diff
@@ -171,33 +185,317 @@ func testSealMigrationShamirToAuto(t *testing.T, setup teststorage.ClusterSetupM
 	verifyBarrierConfig(t, r, wrapping.Shamir, 5, 3, 0)
 }
 
+func TestSealMigration_ShamirToTestSeal(t *testing.T) {
```

Comment:
What's the motivation for these tests?  I've found the test seal to create more confusion and uncertainty than anything else.  I sort had in mind to get rid of it altogether now that we can use transit instead.

Reply:
The motivation for `TestSealMigration_TransitToTestSeal` is that we know e.g. Transit-to-Transit is broken, but we are demostrating that migration works if the auto-seals are of different types.  In our discussion yesterday the consensus was that using  test seal was better than e.g. setting up a local AWS KMS server, for cases where we wanted an auto-unseal option that was not transit.

`TestSealMigration_ShamirToTestSeal` is perhaps redundant -- I included it for completeness, and would be fine if we don't want it, just let me know.  

I do want to keep `TestSealMigration_TransitToTestSeal` though, it plugs a hole in the test suite.

Reply:
I see.  One of my main confusions is that the test seal can be configured to behave as many types of seal.  If we could get rid of TestSealOpts.StoredKeys I'd have less of a problem with it.

For now let's not worry about it, I'm hopeful that by the end of your project the issue will be moot.